core/types/row_consumption.go

```diff
@@ -1,28 +1,18 @@
 package types
 
 import (
-	"fmt"
-	"io"
-
-	"github.com/scroll-tech/go-ethereum/rlp"
+	"github.com/scroll-tech/go-ethereum/common/hexutil"
 )
 
-type RowConsumption struct {
-	Rows uint64
+//go:generate gencodec -type RowConsumptionEntry -field-override rowConsumptionEntryMarshaling -out gen_row_consumption_json.go
+type RowConsumptionEntry struct {
+	Key  string `json:"key" gencodec:"required"`
+	Rows uint64 `json:"rows" gencodec:"required"`
 }
 
-func (rc *RowConsumption) EncodeRLP(w io.Writer) error {
-	return rlp.Encode(w, rc.Rows)
-}
+type RowConsumption []RowConsumptionEntry
 
-func (rc *RowConsumption) DecodeRLP(s *rlp.Stream) error {
-	_, size, err := s.Kind()
-	if err != nil {
-		return err
-	}
-	if size <= 8 {
-		return s.Decode(&rc.Rows)
-	} else {
-		return fmt.Errorf("invalid input size %d for origin", size)
-	}
+// field type overrides for gencodec
+type rowConsumptionEntryMarshaling struct {
+	Rows hexutil.Uint64
```

Comment:
this type definition seems outdated?

Reply:
It's an override.
https://github.com/scroll-tech/go-ethereum/pull/398#discussion_r1267100192